jbpm-human-task/jbpm-human-task-core/src/main/java/org/jbpm/services/task/impl/util/HumanTaskHandlerHelper.java

```diff
@@ -101,36 +102,51 @@ public static Deadlines setDeadlines(Map<String, Object> parameters, List<Organi
 	            String expireComponents = mainComponents[1].substring(1, mainComponents[1].length()-1);
 	 
 	            String[] expireElements = expireComponents.split(ATTRIBUTES_ELEMENTS_SEPARATOR);
-	            Deadline taskDeadline = null;
-	            
+
 	            for (String expiresAt : expireElements) {
-	            	logger.debug("Expires at is {}", expiresAt);
-	                taskDeadline = TaskModelProvider.getFactory().newDeadline();
-	                if (businessCalendar != null) {
-	                	taskDeadline.setDate(businessCalendar.calculateBusinessTimeAsDate(expiresAt));
-	                } else {
-	                	taskDeadline.setDate(new Date(System.currentTimeMillis() + TimeUtils.parseTimeString(expiresAt)));
-	                }
-	                logger.debug("Calculated date of execution is {} and current date {}", taskDeadline.getDate(), new Date());
-	                List<Escalation> escalations = new ArrayList<Escalation>();
-	                
-	                Escalation escalation = TaskModelProvider.getFactory().newEscalation();
-	                escalations.add(escalation);
-	                
-	                escalation.setName("Default escalation");
-	                
-	                taskDeadline.setEscalations(escalations);
-	                escalation.setReassignments(parseReassignment(actionComponent));
-	                escalation.setNotifications(parseNotifications(actionComponent, businessAdministrators));
-	                
-	                deadlines.add(taskDeadline);
+	            	if(businessCalendar != null) {
+						deadlines.add(getNewDeadline(expiresAt, businessCalendar.calculateBusinessTimeAsDate(expiresAt.trim()), actionComponent, businessAdministrators));
+					} else {
+						if (DateTimeUtils.isRepeatable(expiresAt.trim())) {
+							int repeatLimit = getDeadlineRepeatLimit(expiresAt.trim());
+							if(repeatLimit > 0) {
+								for(int i=0; i < repeatLimit; i++) {
+									Date durationDate = getDeadlineDurationDate(expiresAt.trim());
+									deadlines.add(getNewDeadline(expiresAt, durationDate, actionComponent, businessAdministrators));
+								}
+							} else {
+								deadlines.add(getNewDeadline(expiresAt, getDeadlineDurationDate(expiresAt.trim()), actionComponent, businessAdministrators));
```

Comment:
In this case will it be repeated without limit? Do we check anywhere that if the deadline expires if we should create a new one if there is no limit for number of repetitions?

Reply:
in case of just R / it should be refreshed when reschedule is called.

Reply:
OK and will this reschedule stop with a definite number of repetitions? Let's say I have R3/PT1H - every hour, up to 3 times - you generate 3 deadlines at once.  And after the first one is executed, reschedule happens again - it will again add 3 deadlines on top of 2 remainings?

(I have also noticed you deleted tmpDate field which means it will generate 3 identical deadlines now in case of R3)

Reply:
reschedule does not keep any currently existing ones. it refreshes completely

Reply:
OK, so this will be then rescheduled infinitely? Each time there will be 3 deadlines created, but after the first is triggered, another 3 are created again?

Reply:
yeah but in that case of hard-expire you should use explicit start/end dates

Reply:
ok, understand. But in that case we can throw away subsequent calls - i.e. 2nd and 3rd deadline will never be executed. They will be always rewritten by the first deadline.

Reply:
not sure i understand. for example

R2/2019-05-27T13:00:00Z/P1Y2M

creates 2 distinct non recurring deadlines now:

2019-05-27T13:00:00Z/P1Y2M
2020-07-01T15:30:00Z/P1Y2M

Reply:
well it looks in this code:

```
if(repeatLimit > 0) {
    for(int i=0; i < repeatLimit; i++) {
        Date durationDate = getDeadlineDurationDate(expiresAt.trim());
        deadlines.add(getNewDeadline(expiresAt, durationDate, actionComponent, businessAdministrators));
    }
```
that you are reusing the same durationDate over and over again. Every time it will create the same deadline.

Reply:
Opps yes, I didnt check in some changes. Updated and look at added tests in HumanTaskHandlerHelperTest